data_validation/partition_builder.py

```diff
@@ -0,0 +1,263 @@
+            # check kept for explicit calls
+            raise ValueError(f"Unknown Partition Type: {self.args.partition_type}")
+
+        return self.args.partition_type
+
+    def _get_yaml_from_config(self, config_manager: ConfigManager) -> Dict:
+        """Return dict objects formatted for yaml validations.
+
+        Args:
+            config_managers (list[ConfigManager]): List of config manager instances.
+        """
+        yaml_config = {
+            consts.YAML_SOURCE: self.args.source_conn,
+            consts.YAML_TARGET: self.args.target_conn,
+            consts.YAML_RESULT_HANDLER: config_manager.result_handler_config,
+            consts.YAML_VALIDATIONS: [config_manager.get_yaml_validation_block()],
+        }
+        return yaml_config
+
+    def partition_configs(self) -> None:
+        """Takes a list of ConfigManager object and splits each it into multiple
+        ConfigManager objects applying supplied partition logic.
+
+        Returns:
+            None
+        """
+
+        if self.partition_type == "primary_key":
+            partition_filters = self._get_primary_key_partition_filters()
+        elif self.partition_type == "primary_key_mod":
+            # TODO: Add support for Primary_key + Mod
+            raise ValueError(
+                f"Partition Type: '{self.partition_type}' is not supported"
+            )
+        elif self.partition_type == "hash_mod":
+            # TODO: Add support for Hash + Mod
+            raise ValueError(
+                f"Partition Type: '{self.partition_type}' is not supported"
+            )
+
+        self._add_partition_filters_and_store(partition_filters)
+
+    def _get_primary_key_partition_filters(self) -> List[List[str]]:
+        """Generate Partition filters for primary_key type partition for all
+        Configs/Tables.
+
+        Returns:
+            A list of lists of partition filters for each table
+        """
+
+        master_filter_list = []
+
+        # Filter for only first primary key (multi-pk filter not supported)
+        primary_keys = cli_tools.get_arg_list(self.args.primary_keys)
+        primary_key = primary_keys[0]
+
+        for config_manager in self.config_managers:
+
+            validation_builder = ValidationBuilder(config_manager)
+
+            source_partition_row_builder = PartitionRowBuilder(
+                primary_key,
+                config_manager.source_client,
+                config_manager.source_schema,
+                config_manager.source_table,
+                validation_builder.source_builder,
+            )
+
+            target_partition_row_builder = PartitionRowBuilder(
+                primary_key,
+                config_manager.target_client,
+                config_manager.target_schema,
+                config_manager.target_table,
+                validation_builder.target_builder,
+            )
+
+            # accepted_datatypes = ("int64", "int32")
```

Comment:
Should this be uncommented?

Reply:
There is a datatype check below, will remove commented block
```
# If Primary key is non numeric, raise Type Error
if not (source_min.is_integer() and target_min.is_integer()):
    raise TypeError(
        f"Supplied Primary key is not of type Numeric: {primary_key}"
    )
```